### PR TITLE
Improve test coverage for security.config/TestSecret

### PR DESCRIPTION
**Pull request by aliok**
Fixes #

## Proposed Changes

- Improve test coverage for security.config/TestSecret
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: control-plane/pkg/reconciler/sink/auth_provider.go [hunk 1]
```diff
@@ -24,13 +24,16 @@ type SecretLocator struct {
 	*eventing.KafkaSink
 }
 
-func (ks *SecretLocator) SecretName() (string, bool, error) {
+func (ks *SecretLocator) SecretName() (string, bool) {
```
Comment (aliok) [inline control-plane/pkg/reconciler/sink/auth_provider.go:27]: Got rid of the 3rd param here as we never return an error

File: control-plane/pkg/reconciler/sink/auth_provider.go [hunk 2]
```diff
@@ -24,13 +24,16 @@ type SecretLocator struct {
...
-		return "", false, nil
+		return "", false
 	}
-	return ks.Spec.Auth.Secret.Ref.Name, true, nil
+	return ks.Spec.Auth.Secret.Ref.Name, true
 }
 
-func (ks *SecretLocator) SecretNamespace() (string, bool, error) {
-	return ks.Namespace, true, nil
+func (ks *SecretLocator) SecretNamespace() (string, bool) {
```
Comment (aliok) [inline control-plane/pkg/reconciler/sink/auth_provider.go:34]: Same story here

File: control-plane/pkg/security/config_test.go [hunk 3]
```diff
@@ -92,6 +92,29 @@ func TestNewSaramaSecurityOptionFromSecret(t *testing.T) {
 			},
 			wantErr: false,
 		},
+		{
```
Comment (aliok) [inline control-plane/pkg/security/config_test.go:95]: added 2 new test cases

File: control-plane/pkg/security/config_test.go [hunk 4]
```diff
@@ -50,7 +50,7 @@ func (sp *SecretProviderFuncMock) F(ctx context.Context, namespace, name string)
 	return sp.secret, sp.err
 }
 
-func TestNewSaramaSecurityOptionFromSecret(t *testing.T) {
+func TestSecret(t *testing.T) {
```
Comment (aliok) [inline control-plane/pkg/security/config_test.go:53]: Test func name was bad

**Comment by aliok** [comment]
cc @pierDipi @devguyio @matzew

File: control-plane/pkg/security/config_test.go [hunk 5]
```diff
@@ -92,6 +92,29 @@ func TestNewSaramaSecurityOptionFromSecret(t *testing.T) {
...
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: "my-ns",
+						Name:      "my-name",
+					},
+					// No ref to secret here
+					Data: map[string]string{},
+				},
+			},
+			wantSecret: nil,
+			wantErr:    false,
```
Comment (pierDipi) [inline control-plane/pkg/security/config_test.go:116]: can you remove this `wantError`?

File: control-plane/pkg/reconciler/sink/auth_provider.go [hunk 6]
```diff
@@ -24,13 +24,16 @@ type SecretLocator struct {
...
+	if !ks.Spec.HasAuthConfig() {
+		return "", false
+	}
+	return ks.Namespace, true
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/sink/auth_provider.go:38]: what about this?
```suggestion
	return ks.Namespace, ks.Spec.HasAuthConfig()
```

Comment (aliok) [reply control-plane/pkg/reconciler/sink/auth_provider.go:38]: I can revert that one, not a big issue.

I thought:
- If there's no secret specified in the spec, we should also make `SecretNamespace()` return nil, similar to `SecretName()`

In the end, the `SecretName()` is always called before `SecretNamespace()` and the execution stops before `SecretNamespace()` call, but I like consistency.

I don't have a strong opinion and can revert

Comment (aliok) [reply control-plane/pkg/reconciler/sink/auth_provider.go:38]: Oh, you have a suggestion. That's actually better. Committing that now

**Review by pierDipi — APPROVED** [review]

**Comment by pierDipi** [comment]
Thanks!

**Comment by aliok** [comment]
/test pull-knative-sandbox-eventing-kafka-broker-integration-tests

Data plane issue

**Pull request merged**